static/js/vue-cdr-access/src/components/full_record/fileList.vue

```diff
@@ -240,9 +247,79 @@ export default {
             return { markDeleted: markedForDeletion, restricted: restrictedAccess };
         },
 
-        ariaLabelText(row) {
-            return this.$t('full_record.edit_title', { title: row.title });
+        showDropdownList(e) {
+            // Close any currently open dropdowns
+            this.closeDropdownLists(e);
+
+            if (e.target.id.startsWith('dcr-download')) {
+                let drop_down = e.target.parentElement.parentElement.querySelector('.dropdown-menu');
+                if (drop_down !== null) {
+                    drop_down.setAttribute('aria-hidden', 'false');
+                    drop_down.classList.add('show-list');
+                }
+            }
+        },
+
+        closeDropdownLists() {
+            document.querySelectorAll('.show-list').forEach(element => {
+                element.setAttribute('aria-hidden', 'true');
+                element.classList.remove('show-list');
+            });
+        },
+
+        downloadButtonHtml() {
+            if (this.showNonImageDownload) {
+                return `<div class="actionlink download">
+                            <a class="download button action" href="/content/${this.brief_object.id}?dl=true"><i class="fa fa-download"></i> ${this.$t('full_record.download')}</a>
+                        </div>`;
+            } else if (this.showImageDownload) {
+                let html = `<div class="dropdown actionlink download image-download-options">
+                <div class="dropdown-trigger">
+                    <button id="dcr-download-${this.brief_object.id}" class="button download-images" aria-haspopup="true" aria-controls="dropdown-menu">
+                    ${this.$t('full_record.download')} <i class="fas fa-angle-down" aria-hidden="true"></i>
+                    </button>
+                </div>
+                <div class="dropdown-menu table-downloads" id="dropdown-menu" role="menu" aria-hidden="true">
+                    <div class="dropdown-content">`;
+
+                if (this.validSizeOption(800)) {
+                   html += `<a href="${this.imgDownloadLink('800')}" class="dropdown-item">${this.$t('full_record.small') } JPG (800px)</a>`;
+                }
+                if (this.validSizeOption(1600)) {
+                    html += `<a href="${this.imgDownloadLink('1600')}" class="dropdown-item">${this.$t('full_record.medium') } JPG (1600px)</a>`;
+                }
+                if (this.validSizeOption(2500)) {
+                    html += `<a href="${this.imgDownloadLink('2500')}" class="dropdown-item">${this.$t('full_record.large') } JPG (2500px)</a>`;
+                }
+
+                if (this.viewOriginal) {
```

Comment:
viewOriginal permission should be based off the permissions of the file object rather that of the work. I'm not really sure if fileList.vue needs to receive the work objects viewOriginal?